packages/simcore-sdk/tests/integration/test_node_ports_v2_r_clone.py

```diff
@@ -0,0 +1,180 @@
+    # generate file with data
+    file_path = upload_file_dir / file_name
+    file_path.write_text(faker.paragraph(nb_sentences=5))
+    return file_path
+
+
+@pytest.fixture
+def local_file_for_download(upload_file_dir: Path, file_name: str) -> Path:
+    local_file_path = upload_file_dir / f"__local__{file_name}"
+    return local_file_path
+
+
+@pytest.fixture
+def s3_object(project_id: str, node_uuid: str, file_name: str) -> str:
+    s3_path = Path(project_id) / node_uuid / file_name
+    return f"{s3_path}"
+
+
+@pytest.fixture
+async def cleanup_s3(
+    r_clone_settings: RCloneSettings, s3_object: str
+) -> AsyncIterable[None]:
+    yield
+    async with _get_s3_object(r_clone_settings, s3_object) as s3_object:
+        await s3_object.delete()
+
+
+@pytest.fixture
+def mock_update_file_meta_data(mocker: MockerFixture) -> None:
+    async def _raise_error(*args, **kwargs) -> None:
+        raise _TestException()
+
+    mocker.patch(
+        "simcore_sdk.node_ports_v2.r_clone._update_file_meta_data",
+        side_effect=_raise_error,
+    )
+
+
+# UTILS
+
+
+@asynccontextmanager
+async def _get_s3_object(
+    r_clone_settings: RCloneSettings, s3_path: str
+) -> AsyncGenerator["aioboto3.resources.factory.s3.Object", None]:
+    session = aioboto3.Session(
+        aws_access_key_id=r_clone_settings.R_CLONE_S3.S3_ACCESS_KEY,
+        aws_secret_access_key=r_clone_settings.R_CLONE_S3.S3_SECRET_KEY,
+    )
+    async with session.resource(
+        "s3", endpoint_url=r_clone_settings.R_CLONE_S3.endpoint
+    ) as s3:
+        s3_object = await s3.Object(
+            bucket_name=r_clone_settings.R_CLONE_S3.S3_BUCKET_NAME,
+            key=s3_path.lstrip(r_clone_settings.R_CLONE_S3.S3_BUCKET_NAME),
+        )
+        yield s3_object
+
+
+async def _download_s3_object(
+    r_clone_settings: RCloneSettings, s3_path: str, local_path: Path
+):
+    async with _get_s3_object(r_clone_settings, s3_path) as s3_object:
+        await s3_object.download_file(f"{local_path}")
+
+
+def _is_file_present(postgres_db: sa.engine.Engine, s3_object: str) -> bool:
+    with postgres_db.begin() as conn:
+        result = conn.execute(
+            file_meta_data.select().where(file_meta_data.c.file_uuid == s3_object)
+        )
+        result_list = list(result)
+        result_len = len(result_list)
+    assert result_len <= 1, result_list
+    return result_len == 1
+
+
+# TESTS
+
+
+async def test_sync_local_to_s3(
+    r_clone_settings: RCloneSettings,
+    s3_object: str,
+    file_to_upload: Path,
+    local_file_for_download: Path,
+    user_id: int,
+    postgres_db: sa.engine.Engine,
+    cleanup_s3: None,
+) -> None:
+    etag = await r_clone.sync_local_to_s3(
+        r_clone_settings=r_clone_settings,
+        s3_object=s3_object,
+        local_file_path=file_to_upload,
+        user_id=user_id,
+    )
+
+    assert isinstance(etag, str)
+    assert '"' not in etag
```

Comment:
enhancement suggestion: Test (all? Maybe there is a python package providing a list?) forbidden parameters are correctly handled here. At some point, I am sure users will manage to put asian language symbols, left-to-right text and non-printable unicode chars in filenames. In a perfect world, an error is thrown and propagated all the way to the frontend.

This is imho a good option to implement checks for this, we'll be happy down the road